e2e_tests/tests/command/test_tensorboard.py

```diff
@@ -159,3 +161,48 @@ def test_start_tensorboard_for_multi_experiment(tmp_path: Path, secrets: Dict[st
                 break
         else:
             raise AssertionError(f"Did not find {SERVICE_READY} in output")
+
+
+@pytest.mark.slow
+@pytest.mark.e2e_cpu
+def test_start_tensorboard_with_custom_image(tmp_path: Path) -> None:
+    """
+    Start a random experiment configured with the shared_fs backend, start a
+    TensorBoard instance pointed to the experiment with custom image, verify
+    the image has been set, and kill the TensorBoard instance.
+    """
+    with FileTree(tmp_path, {"config.yaml": shared_fs_config(1)}) as tree:
+        config_path = tree.joinpath("config.yaml")
+        experiment_id = exp.run_basic_test(
+            str(config_path), conf.fixtures_path("no_op"), num_trials
+        )
+    command = [
+        "det",
+        "-m",
+        conf.make_master_url(),
+        "tensorboard",
+        "start",
+        str(experiment_id),
+        "--no-browser",
+        "--detach",
+        "--config",
+        "environment.image=alpine",
+    ]
+    env = os.environ.copy()
+    env["DET_DEBUG"] = "true"
+    res = subprocess.run(
+        command, universal_newlines=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE, env=env
+    )
+    assert res.returncode == 0, "\nstdout:\n{} \nstderr:\n{}".format(res.stdout, res.stderr)
+    t_id = res.stdout.replace("\n", "")
+    command = ["det", "-m", conf.make_master_url(), "tensorboard", "config", t_id]
+    res = subprocess.run(
+        command, universal_newlines=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE, env=env
+    )
+    assert res.returncode == 0, "\nstdout:\n{} \nstderr:\n{}".format(res.stdout, res.stderr)
+    config = yaml.safe_load(res.stdout)
+    assert (
+        config["environment"]["image"]["cpu"] == "alpine"
+        and config["environment"]["image"]["cuda"] == "alpine"
+        and config["environment"]["image"]["rocm"] == "alpine"
+    ), "\nSetting custom image not working properly:\n{}".format(config["environment"]["image"])
```

Comment:
Avoid `.format()` in any new code.  There was a time when we supported python 3.5 but now we don't even support 3.6.  Use f-strings instead.

But in this case, just omit the string entirely.  If it fails, people will know this info from the fact that `test_start_tensorboard_with_custom_image` failed.

Also, just include the whole raw output.  That way the failing test logs are as useful as possible ("useful" meaning "devs can understand what failed with just the test logs and the test code that ran"):

```
assert (...), res.stdout
```